lib/middleware/v3/createEntryFileJavaScript.js

```diff
@@ -0,0 +1,47 @@
+'use strict';
+
+const path = require('path');
+const writeFile = require('fs').promises.writeFile;
+const lodash = require('lodash');
+
+/**
+ * @param {string} installationDirectory
+ * @param {Object<string, string>} modules
+ * @param {string} [callback]
+ */
+async function createEntryFileJavaScript(installationDirectory, modules, callback) {
+	const entryFileLocation = path.join(installationDirectory, '/index.js');
+
+	const moduleNames = Array.from(Object.keys(modules)).sort();
+
+	let entryFileContents = '';
+	if (callback) {
+		entryFileContents += 'var modules = {};\n';
```

Comment:
is this a global definition of the variable `modules`?

Reply:
oops, yes, let's use `let` instead of `var` to ensure that `modules` is not attached to the global environment 👍.


```suggestion
		entryFileContents += 'let modules = {};\n';
```